orchagent/portsorch.cpp

```diff
@@ -1161,27 +1161,62 @@ bool PortsOrch::getPortAdminStatus(sai_object_id_t id, bool &up)
     return true;
 }
 
-bool PortsOrch::setPortMtu(sai_object_id_t id, sai_uint32_t mtu)
+bool PortsOrch::getPortMtu(const Port& port, sai_uint32_t &mtu)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+    attr.id = SAI_PORT_ATTR_MTU;
+
+    sai_status_t status = sai_port_api->get_port_attribute(port.m_port_id, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        return false;
+    }
+
+    mtu = attr.value.u32 - (uint32_t)(sizeof(struct ether_header) + FCS_LEN + VLAN_TAG_LEN);
+
+    if (isMACsecPort(port.m_port_id))
+    {
+        mtu -= MAX_MACSEC_SECTAG_SIZE;
+    }
+
+    return true;
+}
+
+bool PortsOrch::setPortMtu(const Port& port, sai_uint32_t mtu)
```

Comment:
Why are we changing this API? Please have it as the original one and have the caller pass the extra value.

Reply:
Due to the Gearbox architecture support, the original function cannot semantically support the functionality of `set MTU`. We just follow other existing functions, like set Speed, set Fec and so on, that needed by Gearbox platform.
https://github.com/sonic-net/sonic-swss/blob/9ec471adc95b9cea1ffcdc6686a875f5ed70f9fe/orchagent/portsorch.cpp#L2191
https://github.com/sonic-net/sonic-swss/blob/9ec471adc95b9cea1ffcdc6686a875f5ed70f9fe/orchagent/portsorch.cpp#L1253

Reply:
ok, from the changes I still think you can reuse the existing function. But I see your point as making it consistent across other `set` functions. Approving from my side.